modules/tensor_mechanics/include/kernels/InertialForce.h

```diff
@@ -10,27 +10,33 @@
 #pragma once
 
 #include "TimeKernel.h"
+#include "ADTimeKernel.h"
 #include "Material.h"
 
 // Forward Declarations
 class TimeIntegrator;
 
-class InertialForce : public TimeKernel
+// parent class
+template <bool is_ad>
+using InertialForceParent = typename std::conditional<is_ad, ADTimeKernel, TimeKernel>::type;
+
+template <bool is_ad>
+class InertialForceTempl : public InertialForceParent<is_ad>
 {
 public:
   static InputParameters validParams();
 
-  InertialForce(const InputParameters & parameters);
+  InertialForceTempl(const InputParameters & parameters);
 
 protected:
-  virtual Real computeQpResidual() override;
+  virtual GenericReal<is_ad> computeQpResidual();
 
-  virtual Real computeQpJacobian() override;
+  virtual Real computeQpJacobian();
 
-  virtual void computeResidualAdditional() override;
+  virtual void computeResidualAdditional();
```

Comment:
why remove the override?

Reply:
Oops. I must have copy pasted these from a similar class (for the  type change). I'll fix. To be clear, we're doing the redundant `virtual ... override` rather than the more elegant `override` without the repeated `virtual`, right? ;-)

Reply:
I m fine with the second one

Reply:
Oh, no, that's not it at all. We simply _cannot_ use override here, as we are deriving from two different classes with different sets of virtual functions!

Reply:
and we are not always overriding. So override should be templated in a way.
ok fine by me!

Reply:
I always use just `override` instead of the silly redundant `virtual ... override` 😄

I'm curious what the compiler says if you leave the `override` on `computeQpResidual()`? Since in that case you are overriding for both instances of the `is_ad` instantiation

Reply:
You'll get an error about a missing override in the `is_ad == false` template instantiation.